ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/plugin/MatchSection.java

```diff
@@ -200,6 +225,23 @@ private void applyMatch(int match) {
 		}
 	}
 
+	private void applyFilter(String filter) {
+		try {
+			if (fCurrentImport == null)
+				return;
+			if (filter == null || filter.isEmpty())
+				((IFeatureImport) fCurrentImport).setFilter(null);
+			else {
+				FrameworkUtil.createFilter(filter);
+				((IFeatureImport) fCurrentImport).setFilter(filter);
+			}
+		} catch (InvalidSyntaxException e) {
+			PDEPlugin.logException(e);
```

Comment:
this should not be logged but create an error marker, so this likely is better done in the Feature Validation process and when users are typing in a filter than during storing of the filter.